lib/l10n_utils/fluent.py

```diff
@@ -5,16 +5,30 @@
 from django.core.cache import caches
 from django.utils.encoding import force_bytes
 
-from fluent.runtime import FluentLocalization, RootedFileResourceLoader
-
-from lib.l10n_utils import translation
+from fluent.runtime import FluentLocalization, FluentResourceLoader
 
 
 cache = caches['l10n']
 
 
+class FluentL10n(FluentLocalization):
+    def has_message(self, message_id):
+        # assume English locales have the message
+        if self.locales[0].startswith('en-'):
+            return True
+
+        for bundle in self._bundles():
+            # bundle.locales[0] is the locale of the loaded FTL files
+            # we want to know if the non-en locale has the message
+            if bundle.locales[0] != 'en' and bundle.has_message(message_id):
+                return True
```

Comment:
I was thinking about doing a `break` here, which could be optional.

And instead of checking for `en`, you could check if `bundle.locales[0] == self.locales[0]`, I think that's generic enough to upstream.

Reply:
Well `return` is another way of saying `break`. That updated check sounds good to me 👍